reference/orders.v2.oas2.yml

```diff
@@ -4147,3 +4262,675 @@ components:
           type: array
           items:
             $ref: '#/components/schemas/shippingAddress_Base'
+  orderCatalogProduct_Post:
+    title: orderCatalogProduct_Post
+    type: object
+    description: |-
+      `product_options` are required if adding a product with variants.
+      Note:
+      - `xxx` and `xxx_customer` always hold the same value, updating either `xxx` or `xxx_customer` will change value for both of those fields. 
+      - If both fields `xxx` and `xxx_customer` are present, they must have same value.
+      - Empty strings `''` and `null` are invalid for `xxx` and `xxx_customer, and `xxx_merchant`.
+      - `xxx`, `xxx_customer`, and `xxx_merchant` default to the value from `catalog` if not supplied in the request.
+    properties:
+      product_id:
+        type: integer
+      name:
+        type: string
+        example: Fog Linen Chambray Towel - Beige Stripe
+        description: Alias for name_customer. The product name that is shown to customer in storefront.
+      name_customer:
+        type: string
+        example: Fog Linen Chambray Towel - Beige Stripe
+        description: The product name that is shown to customer in storefront.
+      name_merchant:
+        type: string
+        example: Towel Type 1
+        description: The product name that is shown to merchant in Control Panel.
+      product_options:
+        type: array
+        items:
+          type: object
+          properties:
+            id:
+              type: integer
+            value:
+              type: string
+            display_name:
+              type: string
+              example: couleur
+              description: Alias for display_name_customer. The product option name that is shown to customer in storefront.
+            display_name_customer:
+              type: string
+              example: couleur
+              description: The product option name that is shown to customer in storefront.
```

Comment:
🍹 Some of these include the text e.g. orderCatalogProduct_Put: line 4391.
> Alias for xyz...
It would be good to have a consistent description for these fields